LEAF_Request_Portal/api/controllers/WorkflowController.php

```diff
@@ -100,65 +102,65 @@ public function post($act)
-			$workflow->setWorkflowID($args[0]);
-			return $workflow->createAction($_POST['stepID'], $_POST['nextStepID'], $_POST['action']);
+			$workflow->setWorkflowID((int)$args[0]);
+			return $workflow->createAction((int)$_POST['stepID'], (int)$_POST['nextStepID'], XSSHelpers::xscrub($_POST['action']));
 		});
 
 		$this->index['POST']->register('workflow/[digit]/step', function($args) use ($workflow) {
-			$workflow->setWorkflowID($args[0]);
-			return $workflow->createStep($_POST['stepTitle'], $_POST['stepBgColor'], $_POST['stepFontColor']);
+			$workflow->setWorkflowID((int)$args[0]);
+			return $workflow->createStep(XSSHelpers::xscrub($_POST['stepTitle']), XSSHelpers::xscrub($_POST['stepBgColor']), XSSHelpers::xscrub($_POST['stepFontColor']));
 		});
 
 		$this->index['POST']->register('workflow/[digit]/initialStep', function($args) use ($workflow) {
-			$workflow->setWorkflowID($args[0]);
-			return $workflow->setInitialStep($_POST['stepID']);
+			$workflow->setWorkflowID((int)$args[0]);
+			return $workflow->setInitialStep((int)$_POST['stepID']);
 		});
 
 		$this->index['POST']->register('workflow/step/[digit]', function($args) use ($workflow) {
-			return $workflow->updateStep($args[0], $_POST['title']);
+			return $workflow->updateStep((int)$args[0], XSSHelpers::xscrub($_POST['title']));
 		});
 
 		$this->index['POST']->register('workflow/step/[digit]/dependencies', function($args) use ($workflow) {
-			return $workflow->linkDependency($args[0], $_POST['dependencyID']);
+			return $workflow->linkDependency((int)$args[0], (int)$_POST['dependencyID']);
 		});
 
 		$this->index['POST']->register('workflow/step/[digit]/indicatorID_for_assigned_empUID', function($args) use ($workflow) {
 			return $workflow->setDynamicApprover($args[0], $_POST['indicatorID']);
 		});
 
 		$this->index['POST']->register('workflow/step/[digit]/indicatorID_for_assigned_groupID', function($args) use ($workflow) {
-			return $workflow->setDynamicGroupApprover($args[0], $_POST['indicatorID']);
+			return $workflow->setDynamicGroupApprover((int)$args[0], (int)$_POST['indicatorID']);
 		});
 
 		$this->index['POST']->register('workflow/dependencies', function($args) use ($workflow) {
-			return $workflow->addDependency($_POST['description']);
+			return $workflow->addDependency(XSSHelpers::xscrub($_POST['description']));
 		});
 
 		$this->index['POST']->register('workflow/dependency/[digit]', function($args) use ($workflow) {
-			return $workflow->updateDependency($args[0], $_POST['description']);
+			return $workflow->updateDependency((int)$args[0], XSSHelpers::xscrub($_POST['description']));
 		});
 
 		$this->index['POST']->register('workflow/dependency/[digit]/privileges', function($args) use ($workflow) {
-			return $workflow->grantDependencyPrivs($args[0], $_POST['groupID']);
+			return $workflow->grantDependencyPrivs((int)$args[0], XSSHelpers::xscrub($_POST['groupID']));
```

Comment:
`groupID` can also be cast to `int` instead of using `xscrub()`.

Reply:
done